pennylane/pulse/hardware_hamiltonian.py

```diff
@@ -266,19 +266,26 @@ def __init__(
         self,
         coeffs,
         observables,
+        reorder_fn=None,
         pulses: List["HardwarePulse"] = None,
         settings: Union["RydbergSettings", "TransmonSettings"] = None,
     ):
         self.settings = settings
         self.pulses = [] if pulses is None else pulses
+        self.reorder_fn = reorder_fn
         super().__init__(coeffs, observables)
 
     def __call__(self, params, t):
-        params = _reorder_parameters(params, self.coeffs_parametrized)
+        if self.reorder_fn is not None:
+            params = self.reorder_fn(params, self.coeffs_parametrized)
         return super().__call__(params, t)
 
     def __add__(self, other):
         if isinstance(other, HardwareHamiltonian):
+            if not self.reorder_fn == other.reorder_fn:
```

Comment:
I think it's a good idea to include a note about parameter reordering functions in the docstring for `HardwareHamiltonian` with information such as making sure that if two hamiltonians are added, their reorder functions must point to the same callable.